### ISSUE TITLE
Error parsing Swagger

### ISSUE THREAD

**Issue by dragonfriend0013** [yarax/swagger-to-graphql#92]
I am using a swagger generated from an IBM tool.  

When parsing this Swagger Doc, i get the following:

```
{
  "errors": [
    {
      "message": "Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESCREDITLIMINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESDATEINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESBALINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESPAYMTINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESPRIMCUSTINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESCUSTDATA\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESSTATCDS_items\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESPDOVRLMITINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESCORPINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESCYCLTDINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESAUTHTODAY\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESPYMTTODAY\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESREFTODAY\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESRETTODAY\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESDECTODAY\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESAVAIL\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESAUTHSTATRESN\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESPRODCODES\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESPROCTYPE\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESCRDWATCH\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"getSup-demo-service_response_200_RESPONSE_RESRESPONSE_RESREALTIMEINFO_RESTIERWTCHGRP\" does not."
    }
  ]
}

```
My Swagger doc is: (i am able to use this Swagger doc to correctly use the API, so i know it is formatted correctly)

```
{
  "swagger" : "2.0",
  "info" : {
    "description" : "",
    "version" : "1.0.0",
    "title" : "sup-demo"
  },
  "host" : "localhost:8080",
  "basePath" : "/sup-demo",
  "schemes" : [ "https", "http" ],
  "consumes" : [ "application/json" ],
  "produces" : [ "application/json" ],
  "paths" : {
    "/{cnum}/{acctid}" : {
      "get" : {
        "tags" : [ "sup-demo" ],
        "operationId" : "getSup-demo-service",
        "parameters" : [ {
          "name" : "Authorization",
          "in" : "header",
          "required" : false,
          "type" : "string"
        }, {
          "name" : "acctid",
          "in" : "path",
          "required" : true,
          "type" : "string"
        }, {
          "name" : "cnum",
          "in" : "path",
          "required" : true,
          "type" : "string"
        } ],
        "responses" : {
          "200" : {
            "description" : "normal response",
            "schema" : {
              "$ref" : "#/definitions/getSup-demo-service_response_200"
            }
          }
        }
      }
    }
  },
  "definitions" : {
    "getSup-demo-service_response_200" : {
      "type" : "object",
      "properties" : {
        "RESPONSE" : {
          "type" : "object",
          "properties" : {
            "RESRESPONSE" : {
              "type" : "object",
              "properties" : {
                "RESTYPECTYPE" : {
                  "type" : "string",
                  "maxLength" : 1
                },
                "RESTPC" : {
                  "type" : "string",
                  "maxLength" : 2
                },
                "RESCPC" : {
                  "type" : "string",
                  "maxLength" : 3
                },
                "RESMULTICARD" : {
                  "type" : "string",
                  "maxLength" : 1
                },
                "RESFAMCRDOPT" : {
                  "type" : "string",
                  "maxLength" : 4
                },
                "RESACSSTRATCD" : {
                  "type" : "string",
                  "maxLength" : 6
                },
                "RESCREDITLIMINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESCREDITLIMIT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999,
                      "maximum" : 99999999999
                    },
                    "RESCASHLIMIT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    }
                  }
                },
                "RESDATEINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESCARDEXPDATE" : {
                      "type" : "string",
                      "maxLength" : 10
                    },
                    "RESDATELASTBILL" : {
                      "type" : "string",
                      "maxLength" : 10
                    },
                    "RESDATEPREVSTMT" : {
                      "type" : "string",
                      "maxLength" : 10
                    },
                    "RESDATEACCTOPN" : {
                      "type" : "string",
                      "maxLength" : 10
                    },
                    "RESDATEACCTCLSD" : {
                      "type" : "string",
                      "maxLength" : 10
                    }
                  }
                },
                "RESBALINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESCURRBAL" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESBALPREVSTMT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESREMSTMTBAL" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESCURRCD" : {
                      "type" : "string",
                      "maxLength" : 3
                    }
                  }
                },
                "RESPAYMTINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESAMTLSTPYMT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESDTLSTPYMT" : {
                      "type" : "string",
                      "maxLength" : 10
                    },
                    "RESSTATEMINPYMT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESDTPAYDUE" : {
                      "type" : "string",
                      "maxLength" : 10
                    },
                    "RESOSMINPAYDUE" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESSTATEOPTSET" : {
                      "type" : "integer",
                      "minimum" : 0,
                      "maximum" : 9999
                    },
                    "RESAMTPRMOGRACE" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESNETPAYPIF" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESREMPROMOGRACE" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    }
                  }
                },
                "RESPRIMCUSTINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESPRIMACCTNUM" : {
                      "type" : "string",
                      "maxLength" : 19
                    },
                    "RESPRIMCUSTID" : {
                      "type" : "integer",
                      "minimum" : 0,
                      "maximum" : 999999999
                    }
                  }
                },
                "RESCUSTDATA" : {
                  "type" : "object",
                  "properties" : {
                    "RESCODE61" : {
                      "type" : "string",
                      "maxLength" : 6
                    },
                    "RESCODE81" : {
                      "type" : "string",
                      "maxLength" : 8
                    }
                  }
                },
                "RESENROLSTATUS" : {
                  "type" : "string",
                  "maxLength" : 1
                },
                "RESLOYIND" : {
                  "type" : "string",
                  "maxLength" : 1
                },
                "RESPROVID" : {
                  "type" : "string",
                  "maxLength" : 6
                },
                "RESSTATCDS" : {
                  "type" : "array",
                  "items" : {
                    "type" : "object",
                    "properties" : {
                      "RESSTATCD" : {
                        "type" : "string",
                        "maxLength" : 2
                      },
                      "RESSTATRSN" : {
                        "type" : "string",
                        "maxLength" : 2
                      }
                    }
                  },
                  "maxItems" : 200,
                  "minItems" : 200
                },
                "RESPDOVRLMITINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESAMTOVLMT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESAMTTOTPASTDUE" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESDTPDROLL" : {
                      "type" : "string",
                      "maxLength" : 10
                    }
                  }
                },
                "RESCURRPRFTCNTR" : {
                  "type" : "string",
                  "maxLength" : 10
                },
                "RESBILLCYL" : {
                  "type" : "integer",
                  "minimum" : 0,
                  "maximum" : 99
                },
                "RESCORPINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESLETHLDCD" : {
                      "type" : "string",
                      "maxLength" : 3
                    },
                    "RESZIPPOSTCD" : {
                      "type" : "string",
                      "maxLength" : 13
                    }
                  }
                },
                "RESCYCLTDINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESAMTDBT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESAMTCRDT" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    },
                    "RESAMTPAYMTS" : {
                      "type" : "number",
                      "format" : "decimal",
                      "minimum" : -99999999999.99,
                      "maximum" : 99999999999.99
                    }
                  }
                },
                "RESPSHIND" : {
                  "type" : "string",
                  "maxLength" : 1
                },
                "RESREALTIMEINFO" : {
                  "type" : "object",
                  "properties" : {
                    "RESAUTHTODAY" : {
                      "type" : "object",
                      "properties" : {
                        "RESNUMAUTHS" : {
                          "type" : "integer",
                          "minimum" : 0,
                          "maximum" : 999
                        },
                        "RESAMTAUTH" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        }
                      }
                    },
                    "RESPYMTTODAY" : {
                      "type" : "object",
                      "properties" : {
                        "RESNUMPYMTS" : {
                          "type" : "integer",
                          "minimum" : 0,
                          "maximum" : 999
                        },
                        "RESAMTPYMTS" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        }
                      }
                    },
                    "RESREFTODAY" : {
                      "type" : "object",
                      "properties" : {
                        "RESNUMREF" : {
                          "type" : "integer",
                          "minimum" : 0,
                          "maximum" : 999
                        },
                        "RESAMTREF" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        }
                      }
                    },
                    "RESRETTODAY" : {
                      "type" : "object",
                      "properties" : {
                        "RESNUMRET" : {
                          "type" : "integer",
                          "minimum" : 0,
                          "maximum" : 999
                        },
                        "RESAMTRET" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        }
                      }
                    },
                    "RESDECTODAY" : {
                      "type" : "object",
                      "properties" : {
                        "RESNUMDEC" : {
                          "type" : "integer",
                          "minimum" : 0,
                          "maximum" : 999
                        },
                        "RESAMTDEC" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        }
                      }
                    },
                    "RESAVAIL" : {
                      "type" : "object",
                      "properties" : {
                        "RESAMTCSH" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        },
                        "RESAMTCSHPAD" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        },
                        "RESAMTMNY" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        },
                        "RESAMTMNYPAD" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        },
                        "RESAVAILREWRDS" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        },
                        "RESTMPAVAL" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        },
                        "RESAMTEXP" : {
                          "type" : "number",
                          "format" : "decimal",
                          "minimum" : -9999999.99,
                          "maximum" : 9999999.99
                        }
                      }
                    },
                    "RESAUTHSTATRESN" : {
                      "type" : "object",
                      "properties" : {
                        "RESSTATUS" : {
                          "type" : "string",
                          "maxLength" : 1
                        },
                        "RESREASON" : {
                          "type" : "string",
                          "maxLength" : 4
                        },
                        "RESREADESC" : {
                          "type" : "string",
                          "maxLength" : 20
                        }
                      }
                    },
                    "RESPRODCODES" : {
                      "type" : "object",
                      "properties" : {
                        "RESCPCCD" : {
                          "type" : "string",
                          "maxLength" : 3
                        },
                        "RESALTCPC" : {
                          "type" : "string",
                          "maxLength" : 3
                        }
                      }
                    },
                    "RESPROCTYPE" : {
                      "type" : "object",
                      "properties" : {
                        "RESPRCSTYP" : {
                          "type" : "string",
                          "maxLength" : 2
                        },
                        "RESALTPRCSTYP" : {
                          "type" : "string",
                          "maxLength" : 2
                        },
                        "RESASSPRCSTYP" : {
                          "type" : "string",
                          "maxLength" : 2
                        }
                      }
                    },
                    "RESCRDWATCH" : {
                      "type" : "object",
                      "properties" : {
                        "RESCWSTATUS" : {
                          "type" : "string",
                          "maxLength" : 4
                        },
                        "RESCWSTARTDT" : {
                          "type" : "string",
                          "maxLength" : 10
                        },
                        "RESCWENDDATE" : {
                          "type" : "string",
                          "maxLength" : 10
                        },
                        "RESCWNEVEND" : {
                          "type" : "string",
                          "maxLength" : 1
                        }
                      }
                    },
                    "RESTIERWTCHGRP" : {
                      "type" : "object",
                      "properties" : {
                        "RESTWSTATUS" : {
                          "type" : "string",
                          "maxLength" : 4
                        },
                        "RESTWSTRTDT" : {
                          "type" : "string",
                          "maxLength" : 10
                        },
                        "RESTWENDDATE" : {
                          "type" : "string",
                          "maxLength" : 10
                        },
                        "RESTWNEVEND" : {
                          "type" : "string",
                          "maxLength" : 1
                        }
                      }
                    }
                  }
                },
                "RESMESSAGE" : {
                  "type" : "string",
                  "maxLength" : 50
                }
              }
            }
          }
        }
      }
    }
  }
}

```

It appears that the parser that is being use has a RegEx that does not allow some special characters in the json, like `-` and `_`.  These are valid json characters so should be supported.

**Comment by h0rac** [comment]
I have same issue, version 1.4

`  "errors": [
    {
      "message": "Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_whatsmyscore\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_customerUsers\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_token\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_isLinked\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_healthcheck\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_whatsmyscore\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_customerUsers\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_token\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_profile_isLinked\" does not.\n\nNames must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"x-swagger-router-controller_healthcheck\" does not."
    }
  ]
}
`

**Comment by 4F2E4A2E** [comment]
graphql-js it self does not allow dashes [1, 2], the message is telling you to use `getSup_demo_service` or  `x_swagger_router_controller_profile` instead.

![image](https://user-images.githubusercontent.com/64590/44953069-b03a4d00-ae8e-11e8-803c-de6ae52fe43d.png)
 
1: https://facebook.github.io/graphql/June2018/#sec-Names
2: https://github.com/graphql/graphql-js/blob/926e4d80c558b107c49e9403e943086fa9b043a8/src/utilities/assertValidName.js#L44

**Comment by h0rac** [comment]
Ok so when I remove x-swagger-route-controller (because swagger .yaml definition do not allow _) how my controller functions will be executed ? also how I can handle authentication for routes

**Comment by yarax** [comment]
Closing since it's not related to the library itself

**Comment by swcho** [comment]
@pojntfx 
I also suffered from using kube's swagger.json.
Though I made PR(https://github.com/yarax/swagger-to-graphql/pull/103) for this, you probably can use custom build for the time being. ;-)
```
$ yarn add https://github.com/swcho/swagger-to-graphql#custom
```

**Comment by cgatian** [comment]
@swcho thank you works beautifully!

@yarax perhaps this could be optionally applied with a config option?

**Comment by 4F2E4A2E** [comment]
> @yarax perhaps this could be optionally applied with a config option?
Config option for what exactly? Please provide an example.

**Comment by yarax** [comment]
Since we don't care much about target names of gql types (they are generated), I think it's completely ok to have it by default, merged and available in 1.4.5
@pojntfx thank you!
